### PR TITLE
Feature/upload save files

### PR DESCRIPTION
**Pull request by xyz8983**
-- model: upload-to attr
-- form: upload multiple files with validation
-- save files to local folder
-- able to check in json file in d3

### CODE REVIEW

**Review by kanastasakis26 — APPROVED** [review]

File: graph_tool/forms.py [hunk 1]
```diff
@@ -2,12 +2,24 @@
 
 class UploadJSONandImage(forms.Form):
     file_field = forms.FileField(
-        # widget=forms.ClearableFileInput(attrs={'multiple': True})
-        error_messages={'required': 'Please upload json file and image files'}
+        widget=forms.ClearableFileInput(attrs={'multiple': True})
```
Comment (boyoonc) [inline graph_tool/forms.py:5]: what is widget variable for?

File: graph_tool/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Generated by Django 2.0.12 on 2019-11-15 22:47
```
Comment (boyoonc) [inline graph_tool/migrations/0001_initial.py:1]: cool!

File: graph_tool/models.py [hunk 3]
```diff
@@ -1,3 +1,5 @@
 from django.db import models
```
Comment (boyoonc) [inline graph_tool/models.py:1]: a django model!!

File: graph_tool/views.py [hunk 4]
```diff
@@ -2,41 +2,38 @@
...
+        if os.path.exists('static/upload_files/'):
+            shutil.rmtree('static/upload_files/')
+        if form.is_valid() and UploadJSONandImage.validate_extension(files.getlist('file_field')):
+            # form cleaned data only check one file instead a list of files, so add extra validation process above
+            for file in files.getlist('file_field'):
+                upload_file_instance = UploadFileField(file_field=file)
+                upload_file_instance.save()
+                if UploadJSONandImage.get_extension(file.name) == 'json':
+                    json_file_url = upload_file_instance.file_field.url
+                    context['json_file_url'] = json_file_url
```
Comment (boyoonc) [inline graph_tool/views.py:33]: nice

**Review by boyoonc — APPROVED** [review]
beautiful

File: graph_tool/forms.py (hunk 1 above)
Comment (xyz8983) [reply graph_tool/forms.py:5]: to allow upload multiple files using the same field

File: graph_tool/models.py (hunk 3 above)
Comment (xyz8983) [reply graph_tool/models.py:1]: yes! so the file can be saved to local folder

**Pull request merged**